x509-cert/src/certificate.rs

```diff
@@ -3,14 +3,43 @@
 use crate::{name::Name, serial_number::SerialNumber, time::Validity};
 use alloc::vec::Vec;
 use const_oid::AssociatedOid;
-use core::cmp::Ordering;
+use core::{cmp::Ordering, fmt::Debug, marker::PhantomData};
 use der::asn1::BitString;
 use der::{Decode, Enumerated, Error, ErrorKind, Sequence, ValueOrd};
 use spki::{AlgorithmIdentifierOwned, SubjectPublicKeyInfoOwned};
 
 #[cfg(feature = "pem")]
 use der::pem::PemLabel;
 
+pub trait ParsingProfile: PartialEq + Debug + Eq + Clone {
```

Comment:
I assume `ParsingProfile` is to disambiguate from other potential usages of the word "profile", but it seems a little unwieldy, especially when all of the generic parameters are named `PP`.

Reply:
Yeah, mostly because I was using `Profile` in the builder PR actually :D
I'll rename that.